src/main/java/manufacturing/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,217 @@
+package manufacturing.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import manufacturing.dao.CarDao;
+import manufacturing.exception.DataProcessingException;
+import manufacturing.lib.Dao;
+import manufacturing.model.Car;
+import manufacturing.model.Driver;
+import manufacturing.model.Manufacturer;
+import manufacturing.util.ConnectionUtil;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars "
+                + "(car_model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car "
+                    + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> getById(Long carId) {
+        String query = "SELECT c.car_id, c.car_model, c.manufacturer_id, "
+                + "m.manufacturer_name, m.manufacturer_country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m "
+                + "ON m.manufacturer_id = c.manufacturer_id "
+                + "WHERE car_id = ? AND c.deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Driver> drivers = createDrivers(carId, connection);
+            while (resultSet.next()) {
+                car = createCar(resultSet, connection);
+                car.setDrivers(drivers);
```

Comment:
- I think it is not necessary to create `drivers` variable - instead you can do like this `car.setDrivers(createDrivers(carId, connection));`
- In order to avoid creation of `Car` with `null` we can do like this `Car car = createCar(resultSet);`, or `return Optional.of(createCar(resultSet));` and if no `Car` found - send `Optional.empty`
- IMHO

Reply:
okay, thanks. took to consideration